### PR TITLE
[#4175] fix(doc): Add more documents to avoid users' misunderstanding

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add more documents to avoid users' misunderstanding

### Why are the changes needed?

Fix: #4175

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Just document.

### CODE REVIEW

**Comment by jerqi** [comment]
@yuqi1129 Could you help me review this pull request?

File: docs/security.md [hunk 1]
```diff
@@ -11,9 +11,11 @@ Apache Gravitino supports two kinds of authentication mechanisms: simple and OAu
 
 ### Simple mode
 
-Simple mode is the default authentication option.
+Simple mode is the default authentication option of the server.
 
-Simple mode allows the client to use the environment variable `GRAVITINO_USER` as the user.
+If the client doesn't set the authentication, the client will use anonymous to access the server.
```
Comment (yuqi1129) [inline docs/security.md:16]: Add this sentence may more proper:

For the client side, if the client doesn't explicitly set the authentication, ...........;

Comment (jerqi) [reply docs/security.md]: OK. I corrected.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**